### PR TITLE
Fix mounting of homeserver.yaml when it does not exist on host

### PR DESCRIPTION
**Pull request by SuperSandro2000**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [ ] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

File: contrib/docker/docker-compose.yml [hunk 1]
```diff
@@ -54,9 +54,13 @@ services:
     image: docker.io/postgres:10-alpine
     # Change that password, of course!
     environment:
+      # we want to create our DB ourselves and this DB gets locked while running the init script
+      - POSTGRES_DB=postgres
       - POSTGRES_USER=synapse
       - POSTGRES_PASSWORD=changeme
     volumes:
+      # DB init script to set LC_COLLATE and LC_CTYPE correctly
+      - ./init-synapse-db.sh:/docker-entrypoint-initdb.d/init-synapse-db.sh
```
Comment (richvdh) [inline contrib/docker/docker-compose.yml]: I believe this is redundant to #6921 ?

File: contrib/docker/docker-compose.yml [hunk 2]
```diff
@@ -15,10 +15,10 @@ services:
     restart: unless-stopped
     # See the readme for a full documentation of the environment settings
     environment:
-      - SYNAPSE_CONFIG_PATH=/etc/homeserver.yaml
+      - SYNAPSE_CONFIG_PATH=/data/homeserver.yaml
```
Comment (richvdh) [inline contrib/docker/docker-compose.yml]: surely this is inconsistent with your mapping to /etc?

File: contrib/docker/docker-compose.yml [hunk 3]
```diff
@@ -15,10 +15,10 @@ services:
...
     volumes:
       # You may either store all the files in a local folder
-      - ./matrix-config/homeserver.yaml:/etc/homeserver.yaml
+      - ./matrix-config/:/etc/
```
Comment (richvdh) [inline contrib/docker/docker-compose.yml]: replacing the whole of /etc is going to break things (notably TLS).

Comment (SuperSandro2000) [reply contrib/docker/docker-compose.yml]: This is not working either as it creates a directory with the name ``homeserver.yaml`` cause the file does not exist when you first run the container.

File: contrib/docker/docker-compose.yml (hunk 2 above)
Comment (SuperSandro2000) [reply contrib/docker/docker-compose.yml]: Oh, you are right. Going to fix that.

File: contrib/docker/docker-compose.yml (hunk 1 above)
Comment (SuperSandro2000) [reply contrib/docker/docker-compose.yml]: It does the same thing in a different way. Going to drop this.

**Comment by SuperSandro2000** [comment]
I dropped the redundant patch and fixed the mount point by dropping it as we already have one on /data.

**Comment by SuperSandro2000** [comment]
If you run ``docker-compose run --rm -e SYNAPSE_SERVER_NAME=my.matrix.host -e SYNAPSE_REPORT_STATS=yes synapse generate`` in ``contrib/docker`` and comment the duplicate traefik labels out you get:
````
tree -d
.
├── files
├── matrix-config
│   └── homeserver.yaml
└── schemas [error opening dir]

\ls matrix-config/ -l
total 4
drwxr-xr-x 2 root root 4096 Feb 27 15:44 homeserver.yaml
````

As you can see the ``homeserver.yaml`` is a directory and not a file cause the file did not exist when the container started to create it.

**Comment by richvdh** [comment]
right, you didn't mention that you were trying to use `generate`. seems sensible to me then.

**Pull request merged**